packages/colorpickers/src/elements/ColorPicker/reducer.ts

```diff
@@ -133,73 +172,136 @@ export const reducer: ReducerType = (state, action) => {
         };
       }
 
-      return state;
+      return {
+        ...state,
+        hexInput: action.payload,
+        color
+      };
     }
     case 'RED_CHANGE': {
-      const red = Number(action.payload);
+      let red = parseInt(action.payload, 10);
+      let color = state.color;
 
-      if (isNaN(red)) return state;
-      if (red > 255) return state;
+      if (!isNaN(red)) {
+        if (red >= 255) {
+          red = 255;
+        }
 
-      const hsl = rgbToHsl(red, state.green, state.blue);
-      const hex = rgbToString(red, state.green, state.blue);
+        if (red < 0) {
+          red = 0;
+        }
+
+        const hsl = rgbToHsl(red, color.green, color.blue);
+        const hex = rgbToString(red, color.green, color.blue);
+
+        color = {
+          ...color,
+          hex,
+          red: action.payload === '' ? color.red : red,
+          hue: hsl.h,
+          lightness: hsl.l,
+          saturation: hsl.s
+        };
+      }
 
       return {
         ...state,
-        hex,
-        red: action.payload === '' ? state.red : red,
-        hue: hsl.h,
-        lightness: hsl.l,
-        saturation: hsl.s
+        redInput: action.payload,
```

Comment:
Input can be updated regardless of whether the temporary color has changed. This matches the 3rd party libraries referenced in the description.